docker/web-and-data/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# This image includes both the web app and the data import related scripts in
+# one image. Because of historic reasons some of these scripts run java
+# artifacts under the hood. It therefore currently makes more sense to have a
+# single image supporting both instead of moving the scripts out to a separate
+# container.
+FROM maven:3.5.4 as build
+COPY $PWD /cbioportal
+WORKDIR /cbioportal
+RUN mvn -DskipTests clean install
+
+FROM shipilev/openjdk-shenandoah:8
```

Comment:
The Docker Hub documentation for this image specifically states that it concerns a nightly builds that is untested, experimental and to be used with care. I would not feel comfortable deploying it to production servers that can affect users and sensitive data. Do the official OpenJDK images, endorsed by a trademarked organisation name, not suffice?

Reply:
@fedde-s for us it doesn't b/c they don't free the memory adequately unfortunately. If you don't care about freeing up memory as much b/c you e.g. have a single machine per app then the official image suffices. Im happy to make this use an official openjdk image though. We'd be using the one in `web/` for production

Reply:
I vote for an official openJDK image.

Reply:
@fedde-s @khzhu I'll update this Dockerfile to the official openJDK image. But I do want to provide one image with Shenandoah cause that's what we are using in production. The performance difference is huge in terms of memory consumption. I could add another `web` only image if that's desired. I get the sensitive data argument but also note that the motto of shiplev is:

`"builds.shipilev.net — still more secure than npm install"`

And we are using the npm registry extensively. Ofc that's not an argument to add another vulnerability. But people/institutions can decide for themselves how much of an issue this is